src/openrct2/object/Object.cpp

```diff
@@ -322,6 +322,29 @@ std::unique_ptr<IStream> ObjectAsset::GetStream() const
     return {};
 }
 
+std::string VersionString(Version version)
+{
+    return String::Format("%i.%i.%i", std::get<0>(version), std::get<1>(version), std::get<2>(version));
+}
+
+Version VersionTuple(std::string version)
+{
+    if (version.length() == 0)
+    {
+        return std::make_tuple(0, 0, 0);
+    }
+
+    auto nums = String::Split(version, ".");
+    uint16_t versions[3]{};
+
+    for (int i = 0; i < 3 && i < nums.size(); i++)
```

Comment:
`String::Split` returns a vector of substrings, why not use that instead of having a hardcoded size of 3? Hotfix releases, for example, could have well more than 3 versions in the version string.

Reply:
That wouldn't be a supported version it should just be x.y.z all other text would be dropped.

Reply:
It's intended that a version like `1.2.3.4.5-somename` would be parsed as `{ 1, 2, 3 }`, and the game woudl see it as equal to `1.2.3.0`? As stated by @IntelOrca [here](https://github.com/OpenRCT2/OpenRCT2/pull/16875#discussion_r835831460), versions like that should be valid (and I agree). With strings, the < operator can still be used to compare versions that use the major/minor/build/revision/etc format.

Reply:
That is true but it fails to compare across decades e.g "5" > "10"

Reply:
> It's intended that a version like `1.2.3.4.5-somename` would be parsed as `{ 1, 2, 3 }`, and the game woudl see it as equal to `1.2.3.0`? As stated by @IntelOrca [here](https://github.com/OpenRCT2/OpenRCT2/pull/16875#discussion_r835831460), versions like that should be valid (and I agree). With strings, the < operator can still be used to compare versions that use the major/minor/build/revision/etc format.

I could not disagree anymore on this. It is a very bad idea to include strings its a complete waste of processing time and would never be used. 1.2.3 is more than enough for any person. Do we know of any object in the history of the game that has had more than like 2 versions let alone 5 levels of versioning.

Reply:
And also how on earth would version replacement work. Is the game magically meant to know that 1.2.3-beta is a higher version than 1.2.3-alpha. I guess we will now need to include a full dictionary of terms for order so that correct replacement can occur.

Reply:
I wasn't suggesting the stuff after the hyphen would be used as part of the comparator, just that it should be part of the storage of the version struct.

Reply:
Where would the string be read?

Reply:
> I wasn't suggesting the stuff after the hyphen would be used as part of the comparator, just that it should be part of the storage of the version struct.

How would that work then that would maximise confusion surely if its there but doesn't do anything.